tests/bin/x-commit-hash/run.t

```diff
@@ -0,0 +1,99 @@
+    > (lang dune 2.4)\
+    > (name whatever)\
+    > EOF
+
+We need to set up a git project for dune-release to work properly
+
+    $ git init > /dev/null
+    $ git add CHANGES.md whatever.opam dune-project README LICENSE
+    $ git commit -m "Initial commit" > /dev/null
+    $ dune-release tag -y
+    [-] Extracting tag from first entry in CHANGES.md
+    [-] Using tag "0.1.0"
+    [+] Tagged HEAD with version 0.1.0
+
+We make a dry-run release
+
+    $ dune-release distrib --dry-run
+    [-] Building source archive
+    => rmdir _build/whatever-0.1.0.build
+    -: exec: git --git-dir .git rev-parse --verify 0.1.0
+    => exec: git --git-dir .git show -s --format=%ct 0.1.0^{commit}
+    => exec: git --git-dir .git clone --local .git _build/whatever-0.1.0.build
+    => exec:
+         git --git-dir _build/whatever-0.1.0.build/.git --work-tree   _build/whatever-0.1.0.build/ checkout --quiet -b dune-release-dist-0.1.0   0.1.0
+    => chdir _build/whatever-0.1.0.build
+       [in _build/whatever-0.1.0.build]
+    -: exec: dune subst
+    -: write whatever.opam
+    => exec: bzip2
+    -: rmdir _build/whatever-0.1.0.build
+    [+] Wrote archive _build/whatever-0.1.0.tbz
+    => chdir _build/
+       [in _build]
+    => exec: tar -xjf whatever-0.1.0.tbz
+    
+    [-] Linting distrib in _build/whatever-0.1.0
+    => chdir _build/whatever-0.1.0
+       [in _build/whatever-0.1.0]
+    => exists ./README
+    [ OK ] File README is present.
+    => exists ./LICENSE
+    [ OK ] File LICENSE is present.
+    => exists ./CHANGES.md
+    [ OK ] File CHANGES is present.
+    => exists whatever.opam
+    [ OK ] File opam is present.
+    -: exec: opam lint -s whatever.opam
+    [ OK ] lint opam file whatever.opam.
+    [ OK ] opam field description is present
+    [ OK ] opam fields homepage and dev-repo can be parsed by dune-release
+    [ OK ] opam field doc can be parsed by dune-release
+    [ OK ] lint _build/whatever-0.1.0 success
+    
+    [-] Building package in _build/whatever-0.1.0
+    => chdir _build/whatever-0.1.0
+    -: exec: dune build -p whatever
+    [ OK ] package builds
+    
+    [-] Running package tests in _build/whatever-0.1.0
+    => chdir _build/whatever-0.1.0
+    -: exec: dune runtest -p whatever
+    [ OK ] package tests
+    -: rmdir _build/whatever-0.1.0
+    
+    [+] Distribution for whatever 0.1.0
+    [+] Commit ...
+    [+] Archive _build/whatever-0.1.0.tbz
+
+We create an opam package:
+
+    $ dune-release opam pkg --dry-run
+    [-] Creating opam package description for whatever
+    dune-release: [WARNING] Could not find _build/whatever-0.1.0.url.
+    dune-release: [WARNING] using https://github.com/foo/whatever/releases/download/0.1.0/whatever-0.1.0.tbz for as url.src. Note that it might differ from the one generated by Github
+    [+] Wrote opam package description _build/whatever.0.1.0/opam
+    $ cat _build/whatever.0.1.0/opam
+    cat: _build/whatever.0.1.0/opam: No such file or directory
```

Comment:
That's weird, this file should exist and contain the x-commit-hash field

Reply:
Hmm yes indeed, any idea what's wrong here? Could you look into it?

Reply:
Maybe try to remove the `--dry-run` argument!

Reply:
good catch

Reply:
although this field contains the commit id and not the tag if there is one, is this what you had in mind for #222 ?

Reply:
That's what I had in mind yes!